File: Data Structures/Stack and Queue/QueueThroughStacks.py
# Queue implementation through stack

class QueueThroughStacks: 
    def __init__(self):
        self.first = []
        self.last = []

    # Peek on the first element
    def peek(self):
        if len(self.last) == 0 and len(self.first) > 0:
            return self.first[0]
        elif len(self.last) > 0 and len(self.first) == 0:
            return self.last[0]
        return None

    # Add an element to the queue 
    def enqueue(self,value):
        l = len(self.first)
        for _ in range(l):
            self.last.append(self.first.pop())
        self.last.append(value)
        return self


    # Remove the first element
    def dequeue(self):
        l = len(self.last)
        for _ in range(l):
            self.first.append(self.last.pop())
        self.first.pop()
        return self


    
# Print outs
my_queue = QueueThroughStacks()
my_queue.peek()
my_queue.enqueue('Joy')
my_queue.enqueue('Matt')
my_queue.enqueue('Pavel')
print(my_queue.peek())
my_queue.dequeue()
print(my_queue.peek())
my_queue.dequeue()
my_queue.dequeue()
print(my_queue.peek())

File: Data Structures/Stack and Queue/QueueThroughStacks.js
// Queue implementation through stack
class QueueThroughStack {
    constructor() {
        this.first = [];
        this.last = [];
    }

    // Peek on the first element
    peek() {
        if (this.last.length === 0 && this.first.length > 0) {
            return this.first[0];
        }
        if (this.first.length === 0 && this.last.length > 0) {
            return this.first[0];
        }
        return null;
    }

    // Add an element to the queue
    enqueue(value) {
        const length = this.first.length;
        for (let i = 0; i < length; i++) {
            this.last.push(this.first.pop());
        }
        this.last.push(value);
        return this;
    }

    // Remove the first element
    dequeue() {
        const length = this.last.length;
        for (let i = 0; i < length; i++) {
            this.first.push(this.last.pop());
        }
        this.first.pop();
        return this;
    }
}

const myQueue = new QueueThroughStack();
myQueue.peek();
myQueue.enqueue('Joy');
myQueue.enqueue('Matt');
myQueue.enqueue('Pavel');
myQueue.peek();
myQueue.dequeue();
myQueue.dequeue();
myQueue.dequeue();
myQueue.peek();